presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupId.java

```diff
@@ -46,7 +46,7 @@ public ResourceGroupId(ResourceGroupId parent, String name)
         return result;
     }
 
-    private ResourceGroupId(List<String> segments)
+    public ResourceGroupId(@JsonProperty("segments") List<String> segments)
```

Comment:
Let's make the JSON representation simply be an array, not an object with a single array field.